File: src/main/java/com/github/alexandrecarlton/idea/settings/applier/impl/common/FileTypeMapper.java
package com.github.alexandrecarlton.idea.settings.applier.impl.common;

import com.intellij.openapi.fileTypes.FileType;
import java.util.function.Supplier;
import javax.inject.Inject;
import javax.inject.Named;

/**
 * Maps our {@link FileType} layout to an IntelliJ file type.
 * If none can be found, a sentinel value is returned.
 */
public class FileTypeMapper {

  private final Supplier<FileType> javascriptFileTypeSupplier;

  @Inject
  public FileTypeMapper(@Named("JavaScript") Supplier<FileType> javascriptFileTypeSupplier) {
    this.javascriptFileTypeSupplier = javascriptFileTypeSupplier;
  }

  public FileType mapFileType(com.github.alexandrecarlton.idea.settings.layout.common.FileType fileType) {
    switch (fileType) {
      case JAVASCRIPT: return javascriptFileTypeSupplier.get();
      default:
        throw new IllegalArgumentException("Unhandled file type: " + fileType);
    }
  }

}

File: src/main/java/com/github/alexandrecarlton/idea/settings/applier/impl/common/FileTypeMapper.kt
package com.github.alexandrecarlton.idea.settings.applier.impl.common

import com.github.alexandrecarlton.idea.settings.layout.common.FileType.JAVASCRIPT
import com.intellij.openapi.fileTypes.FileType
import java.util.function.Supplier
import javax.inject.Inject
import javax.inject.Named

/**
 * Maps our [FileType] layout to an IntelliJ file type.
 * If none can be found, a sentinel value is returned.
 */
class FileTypeMapper @Inject
constructor(@param:Named("JavaScript") private val javascriptFileTypeSupplier: Supplier<FileType>) {

    fun mapFileType(fileType: com.github.alexandrecarlton.idea.settings.layout.common.FileType) =
        when (fileType) {
            JAVASCRIPT -> javascriptFileTypeSupplier.get()
        }
}
